### PR TITLE
Add Exercise 3: Reverse Temperature Converter for Chapter 1.3 (#144)

### LINKED ISSUE #144: Exercise 3: The for Statement ("for_statement") ("Chapter 1.3")
Looking for help creating exercise:
Topic: **The for Statement**
Exercise Directory Name: **for_statement**

Note: For Directory Name, refer to the exercises directory to find the correct number to use. It should be close to the number in the issue title but may be slightly off.

Refer to "The C Programming Language" book, Chapter 1.3 for more information.

Contribution instructions: https://github.com/danwritecode/clings?tab=readme-ov-file#contributions

### PR DESCRIPTION
**Pull request by 7etsuo**
### Description

This pull request adds two new exercises to help learners understand and practice using `for` loops in C programming.

### Changes

1. **Exercise 1 (`for_statement_01.c`)**:
   - Converts a `while` loop that prints numbers from 0 to 9 into a `for` loop.
   - Includes assertions to verify correct loop execution and counting.

2. **Exercise 2 (`for_statement_02.c`)**:
   - Modifies a temperature conversion program to print a Fahrenheit to Celsius conversion table in reverse order, from 300 degrees to 0.
   - Uses a `for` loop decrementing from 300 by 20 each iteration.
   - Includes assertions to verify the reversal steps of the temperature.

NOTE: We start at 300 and go down to zero, then have the user start at 0 and go up to 300. This approach reverses the sequence presented in the book. My reasoning for this is to avoid the negative number in the assert() function, although I could simply add 20 to adjust it.

### Testing

Both exercises have been tested locally:
- Exercise 1 correctly prints numbers from 0 to 9 and passes the assertion tests.
- Exercise 2 correctly prints the temperature table in reverse order and passes its assertions.

resolves #144

### CODE REVIEW

File: exercises/03_for_statement/for_statement_01.c [hunk 1]
```diff
@@ -0,0 +1,66 @@
...
+ * 		...
+ *		increment;
+ *	}
+ */
+
+/** EXERCISE
+ * Your job is to write a for statement equivalent to the following 
+ * while statement:
+ *
+ * i = 0;								// initialization
```
Comment (7etsuo) [inline exercises/03_for_statement/for_statement_01.c:27]: this needs to be fixed.

File: exercises/03_for_statement/for_statement_01.c [hunk 2]
```diff
@@ -0,0 +1,66 @@
...
+ * while (i < 10) {			// condition
+ *	printf("%d\n", i);
+ *	verify_count(&count, i);
+ *	i++;								// increment
```
Comment (7etsuo) [inline exercises/03_for_statement/for_statement_01.c:31]: This comment is also bad

**Comment by danwritecode** [comment]
resolves #144

**Review by danwritecode — APPROVED** [review]
These are great exercises, raised that bar above what I did on 02. This is the new quality standard!

**Pull request merged**